File: 0454-4sum-ii/0454-4sum-ii.java
class Solution {
    public int fourSumCount(int[] nums1, int[] nums2, int[] nums3, int[] nums4) {
        Map<Integer,Integer>mpp=new HashMap<>();

        for(int x:nums1){
            for(int y:nums2){
                mpp.put(x+y,mpp.getOrDefault(x+y,0)+1);
            }
        }

        int ans=0;
        for(int x:nums3){
            for(int y:nums4){
                ans+=mpp.getOrDefault(-(x+y),0);
            }
        }
        return ans;
    }
}

File: 0454-4sum-ii/0454-4sum-ii.cpp
class Solution {
public:
    int fourSumCount(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3, vector<int>& nums4) {
        unordered_map<int,int>mpp;

        for(auto x:nums1){
            for(auto y:nums2){
                mpp[x+y]++;
            }
        }

        int ans=0;
        for(auto x:nums3){
            for(auto y:nums4){
                ans+=mpp[-(x+y)];
            }
        }
        return ans;
    }
};